src/main.ts

```diff
@@ -0,0 +1,656 @@
+    github: {
+      clientID: values.GITHUB_OAUTH_CLIENT_ID,
+      secret: values.GITHUB_OAUTH_CLIENT_SECRET,
+    },
+  },
+};
+
+const openai = new OpenAI({
+  apiKey: values.OPENAI_API_KEY,
+});
+
+// epoch - milliseconds
+export const map = new Map<
+  string,
+  { accessToken: string; expiresEpoch: number }
+>();
+
+const address = new URL("http://localhost:4000");
+const GITHUB_CALLBACK = new URL("/auth/github/callback", address);
+
+const authz = {
+  github: auth.GitHub({
+    clientID: config.oauth.github.clientID,
+    clientSecret: config.oauth.github.secret,
+    scope: "user:email read:user",
+    redirectURI: GITHUB_CALLBACK.toString(),
+    logger: console,
+  }),
+};
+
+export function getGitHubClient() {
+  const client = Octokit.plugin(paginateGraphQL);
+  return client;
+}
+
+const app = new Hono();
+
+const db = new Database("db.sqlite");
+
+db.query(
+  `create table if not exists "tokens" ( user_id text primary key, refresh_token text, expiration_date number, user_name string );`
+).all();
+
+app.get("/", (c) => {
+  const userID = c.req.query("user_id") || `u_${nanoid()}`;
+
+  return c.html(`
+		<body>
+			<p> Your user id is ${userID} </p>
+			<a href="/auth/github/authorize?user_id=${userID}"> Link GitHub </a>
+      &nbsp
+			<a href="/tokens?user_id=${userID}"> See my tokens </a>
+      &nbsp
+      <a href="/commits?user_id=${userID}"> See my commits </a>
+      &nbsp
+      <a href="/PRs?user_id=${userID}"> See my PRs </a>
+      &nbsp
+      <a href="/summary?user_id=${userID}"> Summary of what I did today! </a>
+		</body>
+		`);
+});
+
+app.get("/auth/github/authorize", async (c) => {
+  const userID = c.req.query("user_id");
+
+  if (!userID) {
+    return c.json("Unauthorized", 401);
+  }
+  const { state, url } = await authz.github.authorize();
+
+  const secure = c.req.url.startsWith("https://");
+
+  const value = pipe(state, JSON.stringify, stringToBase64);
+  setCookie(c, "state", value, {
+    maxAge: 10 * seconds.minute,
+    httpOnly: true,
+    ...(secure ? { secure: secure, sameSite: "none" } : {}),
+  });
+  const valueUser = pipe(userID, JSON.stringify, stringToBase64);
+  setCookie(c, "user", valueUser, {
+    maxAge: 10 * seconds.minute,
+    httpOnly: true,
+    ...(secure ? { secure: secure, sameSite: "none" } : {}),
+  });
+
+  return c.redirect(url.toString());
+});
+
+app.get("/auth/github/callback", async (c) => {
+  const cookie = getCookie(c, "state");
+  const userID = getCookie(c, "user");
+
+  if (!userID) {
+    return c.json({ message: "unauthorized" }, { status: 401 });
+  }
+
+  if (!cookie) {
+    return c.json({ message: "No state" }, { status: 400 });
+  }
+
+  const state = pipe(cookie, base64ToString, JSON.parse);
+  const response = await authz.github.callback(
+    new URL(c.req.url),
+    state,
+    GITHUB_CALLBACK.toString()
+  );
+
+  const currentTimestamp = Date.now();
+
+  const endDate =
+    currentTimestamp + (response.refresh_token_expires_in as number) * 1000;
```

Comment:
endDate is a bad name for the variable, endDate of what?

refreshTokenExpirationTimestamp is a bit long, but very clear imo